app/controllers/application_controller.rb

```diff
@@ -72,4 +72,11 @@ def encryptor
   def metrics_logger
     METRICS_LOGGER
   end
+
+  def ensure_visit_integrity
+    unless visit && visit.prisoner && visit.prisoner.prison_name.present?
+      redirect_to edit_prisoner_details_path,
+        notice: 'We’re missing some vital information'
```

Comment:
Thanks for running by me @SteveMarshall. Is this like an error? Suggest we should say something like: 
++ 
You need to enter the prisoner's name